lib/tasks/whitehall_importer.rb

```diff
@@ -164,6 +174,10 @@ def set_published_state_first?(state)
       state == "withdrawn"
     end
 
+    def published_author(whitehall_edition)
+      whitehall_edition["revision_history"].select { |h| h["state"] == "published" }.first
```

Comment:
I think you'd want `.last` instead of `.first` here as an edition can be published and the unpublished multiple times on the same edition.